hydrator-common/src/main/java/co/cask/hydrator/common/SchemaValidator.java

```diff
@@ -29,33 +30,39 @@
 public final class SchemaValidator {
 
   /**
-   * Validate output schema fields and if input schema is present
+   * Validate output schema fields and if input schema is present,
    * check if output schema is a subset of the input schema and return output schema.
    * @param outputSchemaString output schema from config
    * @param rowKeyField row key field from config
    * @param pipelineConfigurer Pipelineconfigurer that's used to get input schema and set output schema.
    * @return Schema - output schema
    */
+  @Nullable
   public static Schema validateOutputSchemaAndInputSchemaIfPresent(String outputSchemaString, String rowKeyField,
-                                                                 PipelineConfigurer pipelineConfigurer) {
-    Schema outputSchema;
-    // initialize output schema if presnet
-    Preconditions.checkNotNull(outputSchemaString, "Output schema string is null");
-
-    try {
-      outputSchema = Schema.parseJson(outputSchemaString);
-    } catch (IOException e) {
-      throw new IllegalArgumentException("Unable to parse output schema : " + e.getMessage(), e);
+                                                                   PipelineConfigurer pipelineConfigurer) {
+    Schema inputSchema = pipelineConfigurer.getStageConfigurer().getInputSchema();
+    if (inputSchema == null && outputSchemaString == null) {
+       return null;
     }
 
-    Schema inputSchema = pipelineConfigurer.getStageConfigurer().getInputSchema();
+    // initialize output schema if present; otherwise, set it to input schema
+    Schema outputSchema;
+    if (outputSchemaString == null) {
+      outputSchema = inputSchema;
```

Comment:
@albertshau should we return null here? as we are setting output schema to be input schema and that get's propagated to the next stage.

but due to CDAP-4575, that's not actually true and in initialize output schema would still be null.